e2e/fixtures/render-type/src/entries.tsx

```diff
@@ -3,27 +3,33 @@ import { createPages } from 'waku/router/server';
 import { ServerEcho } from './ServerEcho.js';
 import { ClientEcho } from './ClientEcho.js';
 
-export default createPages(async ({ createPage }) => {
-  createPage({
-    render: 'dynamic',
-    path: '/server/dynamic/[echo]',
-    component: ServerEcho,
-  });
-  createPage({
-    render: 'static',
-    path: '/server/static/[echo]',
-    staticPaths: ['static-echo'],
-    component: ServerEcho,
-  });
-  createPage({
-    render: 'dynamic',
-    path: '/client/dynamic/[echo]',
-    component: ClientEcho,
-  });
-  createPage({
-    render: 'static',
-    path: '/client/static/[echo]',
-    staticPaths: ['static-echo'],
-    component: ClientEcho,
-  });
-});
+// This needs type annotations for the return type of createPages
+// @see https://github.com/microsoft/TypeScript/issues/42873#issuecomment-2065572017
```

Comment:
I was curious about this explicit annotation in other repo.
Does the linked issue commend suggest the explicit annotation?

Reply:
Yes... it is apparently a known issue with type inference inside pnpm projects with symlinked packages.

After visiting the discord for [arktype](https://github.com/arktypeio/arktype) I was linked to that issue on the Typescript repo that I left in the comments. Here is the summary that I ended up putting on the arktype discord after a chat with their author:

![image](https://github.com/user-attachments/assets/509a17e0-1a7a-4504-87d3-795cc20ddd9e)

[discord link if you'd like to read the full thread](https://discord.com/channels/957797212103016458/1282402037731950653/1282402037731950653)

[arktype discord if you need a join link](https://arktype.io/discord)